src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.cpp

```diff
@@ -0,0 +1,765 @@
+    auto min_entry_result = type_registry_entries_.insert({type_ids.type_identifier1(), minimal_entry});
+    auto max_entry_result = type_registry_entries_.insert({type_ids.type_identifier2(), complete_entry});
+    if (!type_ids_result.second || !min_entry_result.second || !max_entry_result.second)
+    {
+        if (local_type_identifiers_[type_name] != type_ids ||
+                type_registry_entries_[type_ids.type_identifier1()] != minimal_entry ||
+                type_registry_entries_[type_ids.type_identifier2()] != complete_entry)
+        {
+            if (type_ids_result.second)
+            {
+                local_type_identifiers_.erase(type_name);
+            }
+            if (min_entry_result.second)
+            {
+                type_registry_entries_.erase(type_ids.type_identifier1());
+            }
+            if (max_entry_result.second)
+            {
+                type_registry_entries_.erase(type_ids.type_identifier2());
+            }
+            return ReturnCode_t::RETCODE_BAD_PARAMETER;
+        }
+    }
+    return ReturnCode_t::RETCODE_OK;
+}
+
+ReturnCode_t TypeObjectRegistry::register_type_identifier(
+        const std::string& type_name,
+        const TypeIdentifier& type_identifier)
+{
+    // Preconditions
+    if (TypeObjectUtils::is_direct_hash_type_identifier(type_identifier) || type_name.empty())
+    {
+        return ReturnCode_t::RETCODE_PRECONDITION_NOT_MET;
+    }
+#if !defined(NDEBUG)
+    try
+    {
+        TypeObjectUtils::type_identifier_consistency(type_identifier);
+    }
+    catch (eprosima::fastdds::dds::xtypes1_3::InvalidArgumentError& exception)
+    {
+        EPROSIMA_LOG_ERROR(XTYPES_TYPE_REPRESENTATION, "Inconsistent TypeIdentifier: " << exception.what());
+        return ReturnCode_t::RETCODE_PRECONDITION_NOT_MET;
+    }
+#endif // !defined(NDEBUG)
+    TypeIdentifierPair type_identifiers;
+    type_identifiers.type_identifier1(type_identifier);
+    auto result = local_type_identifiers_.insert({type_name, type_identifiers});
+    if (!result.second)
+    {
+        if (local_type_identifiers_[type_name] != type_identifiers)
+        {
+            return ReturnCode_t::RETCODE_BAD_PARAMETER;
+        }
+    }
+    return ReturnCode_t::RETCODE_OK;
+}
+
+ReturnCode_t TypeObjectRegistry::get_type_objects(
+        const std::string& type_name,
+        TypeObjectPair& type_objects)
+{
+    if (type_name.empty())
+    {
+        return ReturnCode_t::RETCODE_PRECONDITION_NOT_MET;
+    }
+    TypeIdentifierPair type_ids;
+    ReturnCode_t ret_code = get_type_identifiers(type_name, type_ids);
+    if (ReturnCode_t::RETCODE_OK == ret_code)
+    {
+        if (!TypeObjectUtils::is_direct_hash_type_identifier(type_ids.type_identifier1()) ||
+                !TypeObjectUtils::is_direct_hash_type_identifier(type_ids.type_identifier2()))
+        {
+            return ReturnCode_t::RETCODE_BAD_PARAMETER;
+        }
+        if (EK_MINIMAL == type_ids.type_identifier1()._d())
```

Comment:
Given the name `complete_to_minimal`, I assumed that 1 was the COMPLETE and 2 the MINIMAL.
I can not find any documentation about it, so i guess the order is random?